qiskit/pulse/channels.py

```diff
@@ -140,6 +140,11 @@ def name(self) -> str:
         """Return the shorthand alias for this channel, which is based on its type and index."""
         return f"{self.__class__.prefix}{self._index}"
 
+    @property
+    def is_schedulable(self) -> bool:
```

Comment:
Rather than having this attribute, I would define `ClassicalIOChannel` class which is a counterpart of `PulseChannel`, and turn `MemorySlot`, `RegisterSlot`,  and `SnapshotChannel` into its subclass. Note that we schedule instructions rather than channels.